apps/science/src/main.cpp

```diff
@@ -1,19 +1,144 @@
+#include "ActuatorController.h"
+#include "CANBuffer.h"
 #include "CANBus.h"
 #include "CANMsg.h"
+#include "Encoder.h"
+#include "EncoderAbsolute_PWM.h"
+#include "EncoderRelative_Quadrature.h"
+#include "LimServo.h"
+#include "MoistureSensor.h"
+#include "ScienceConfig.h"
 #include "hw_bridge.h"
 #include "mbed.h"
 
-DigitalOut led1(LED1);
+// motors
+Motor elevatorMotor(MTR_PWM_2, MTR_DIR_2, false);
+Motor indexerMotor(MTR_PWM_1, MTR_DIR_1, false);
+
+// Servo
+LimServo coverServo(SRVO_PWM_1, ScienceConfig::coverServoRange, ScienceConfig::coverServoMaxPulse,
+                    ScienceConfig::coverServoMinPulse);
+
+LimServo diggerServo(SRVO_PWM_2, ScienceConfig::diggerServoRange, ScienceConfig::diggerServoMaxPulse,
+                     ScienceConfig::diggerServoMinPulse);
+
+// encoders
+EncoderAbsolute_PWM elevatorEncoder(ScienceConfig::elevatorEncoderConfig);
+EncoderRelative_Quadrature indexerEncoder(ScienceConfig::centrifugeEncoderConfig);
 
-constexpr auto k_interval = 500ms;
+// limit switches
+DigitalIn elevatorLimTop(LIM_SW_3);
+DigitalIn elevatorLimBottom(LIM_SW_4);
+DigitalIn indexerLimFront(LIM_SW_1);
+DigitalIn indexerLimBack(LIM_SW_2);
+
+// Actuators
+ActuatorController elevatorActuator(ScienceConfig::diggerLiftActuatorConfig, elevatorMotor, elevatorEncoder,
+                                    elevatorLimBottom, elevatorLimTop);
+ActuatorController indexerActuator(ScienceConfig::indexerActuatorConfig, indexerMotor, indexerEncoder, indexerLimFront,
+                                   indexerLimBack);
+
+// I2C
+MoistureSensor moistureSensor = MoistureSensor(TEMP_MOIST_I2C_SDA, TEMP_MOIST_I2C_SCL);
+
+// LED used for debugging
+DigitalOut led1(LED1);
 
+// Declaring CANBus object
 CANBus can(CAN1_RX, CAN1_TX, HWBRIDGE::ROVERCONFIG::ROVER_CANBUS_FREQUENCY);
 
+// static objects
+static mbed_error_status_t setMotionData(CANMsg &msg) {
+  float motionData;
+  msg.getPayload(motionData);
+
+  switch (msg.getID()) {
+    case HWBRIDGE::CANID::SET_GENEVA_INDEX:
```

Comment:
yes. great catch. @Yehia-Fahmy you need to change the functionality st it simply sets the angle. also you need to rename this can id. the way to do this is to rename is to create a PR in the hw bridge repo that changes the name to SET_GENEVA_ANGLE and merge that in and then update your submodule

Reply:
i opened a PR in the hw_bridge repo and changed the function being called in this case

Reply:
specifically, i changed setMotionData to setAngle_Degrees